tests/Rules/Doctrine/ORM/EntityColumnRuleTest.php

```diff
@@ -271,12 +271,26 @@ public function testSuperclass(?string $objectManagerLoader): void
 	{
 		$this->allowNullablePropertyForRequiredField = false;
 		$this->objectManagerLoader = $objectManagerLoader;
-		$this->analyse([__DIR__ . '/data/MyBrokenSuperclass.php'], [
-			[
-				'Property PHPStan\Rules\Doctrine\ORM\MyBrokenSuperclass::$five type mapping mismatch: database can contain resource but property expects int.',
-				17,
-			],
-		]);
+
+		$dbalVersion = InstalledVersions::getVersion('doctrine/dbal');
+		$hasDbal4 = $dbalVersion !== null && strpos($dbalVersion, '4.') === 0;
+		if ($hasDbal4) {
+			$errors = [
+				[
+					'Property PHPStan\Rules\Doctrine\ORM\MyBrokenSuperclass::$five type mapping mismatch: database can contain string but property expects int.',
```

Comment:
Instead of putting the entire `$errors` assignment in an if, do the assignment once and only change the part of the string that changes with `sprintf` and ternary operator.